[BUG]: Task :react-native-video:compileDebugKotlin FAILED

### Version

6.15.0

### What platforms are you having the problem on?

Android

### System Version

Mac Os - 15.3.2 (24D81) - M1 Chip

### On what device are you experiencing the issue?

Simulator

### Architecture

New architecture with interop layer

### What happened?

 /node_modules/react-native-video/android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt:293:53 Type argument is not within its bounds: must be subtype of 'Event<Event<*>>'.
/node_modules/react-native-video/android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt:293:66 Argument type mismatch: actual type is 'Int', but 'Int' was expected.
/node_modules/react-native-video/android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt:293:77 Argument type mismatch: actual type is 'Int', but 'Int' was expected. FAILURE: Build failed with an exception.  ------- 
----------------------------------------------------------------------------------------------------------------
Encountering these errors in react native version : 0.80.0
React Native Video - 6.15.0 

### Reproduction Link

repository link

### Reproduction

Step to reproduce this bug are: 
Install latest version (6.15.0) 

in the app repo

cd android && ./gradlew clean && cd .. && npx react-native run-android

--- a/android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt
+++ b/android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt
@@ -288,12 +288,7 @@ class VideoEventEmitter {
         }
     }
 
-    private class VideoCustomEvent(
-        surfaceId: Int,
-        viewId: Int,
-        private val event: EventTypes,
-        private val paramsSetter: (WritableMap.() -> Unit)?
-    ) :
+    private class VideoCustomEvent(surfaceId: Int, viewId: Int, private val event: EventTypes, private val paramsSetter: (WritableMap.() -> Unit)?) :
         Event<VideoCustomEvent>(surfaceId, viewId) {
 
         override fun getEventName(): String = "top${event.eventName.removePrefix("on")}"
@@ -301,11 +296,7 @@ class VideoEventEmitter {
         override fun getEventData(): WritableMap? = Arguments.createMap().apply(paramsSetter ?: {})
     }
 
-    private class EventBuilder(
-        private val surfaceId: Int,
-        private val viewId: Int,
-        private val dispatcher: EventDispatcher
-    ) {
+    private class EventBuilder(private val surfaceId: Int, private val viewId: Int, private val dispatcher: EventDispatcher) {
         fun dispatch(event: EventTypes, paramsSetter: (WritableMap.() -> Unit)? = null) =
             dispatcher.dispatchEvent(VideoCustomEvent(surfaceId, viewId, event, paramsSetter))
     }

--- a/android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt
+++ b/android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt
@@ -288,12 +288,25 @@ class VideoEventEmitter {
         }
     }
 
+    private class VideoCustomEvent(
+        surfaceId: Int,
+        viewId: Int,
+        private val event: EventTypes,
+        private val paramsSetter: (WritableMap.() -> Unit)?
+    ) : Event<VideoCustomEvent>(surfaceId, viewId) {
+
+        override fun getEventName(): String {
+            return "top${event.eventName.removePrefix("on")}"
+        }
+
+        override fun getEventData(): WritableMap? {
+            return Arguments.createMap().apply(paramsSetter ?: {})
+        }
+    }
+
     private class EventBuilder(private val surfaceId: Int, private val viewId: Int, private val dispatcher: EventDispatcher) {
         fun dispatch(event: EventTypes, paramsSetter: (WritableMap.() -> Unit)? = null) =
-            dispatcher.dispatchEvent(object : Event<Event<*>>(surfaceId, viewId) {
-                override fun getEventName() = "top${event.eventName.removePrefix("on")}"
-                override fun getEventData() = Arguments.createMap().apply(paramsSetter ?: {})
-            })
+            dispatcher.dispatchEvent(VideoCustomEvent(surfaceId, viewId, event, paramsSetter))
     }
 
     private fun audioTracksToArray(audioTracks: java.util.ArrayList<Track>?): WritableArray =

--- a/android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt
+++ b/android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt
@@ -293,18 +293,19 @@ class VideoEventEmitter {
         viewId: Int,
         private val event: EventTypes,
         private val paramsSetter: (WritableMap.() -> Unit)?
-    ) : Event<VideoCustomEvent>(surfaceId, viewId) {
+    ) :
+        Event<VideoCustomEvent>(surfaceId, viewId) {
 
-        override fun getEventName(): String {
-            return "top${event.eventName.removePrefix("on")}"
-        }
+        override fun getEventName(): String = "top${event.eventName.removePrefix("on")}"
 
-        override fun getEventData(): WritableMap? {
-            return Arguments.createMap().apply(paramsSetter ?: {})
-        }
+        override fun getEventData(): WritableMap? = Arguments.createMap().apply(paramsSetter ?: {})
     }
 
-    private class EventBuilder(private val surfaceId: Int, private val viewId: Int, private val dispatcher: EventDispatcher) {
+    private class EventBuilder(
+        private val surfaceId: Int,
+        private val viewId: Int,
+        private val dispatcher: EventDispatcher
+    ) {
         fun dispatch(event: EventTypes, paramsSetter: (WritableMap.() -> Unit)? = null) =
             dispatcher.dispatchEvent(VideoCustomEvent(surfaceId, viewId, event, paramsSetter))
     }

